packages/liquidator/src/OneInchExchange.js

```diff
@@ -59,38 +84,55 @@ class OneInchExchange {
       });
     }
 
-    // 1 Split config
-    const flags = 0; // Enables all exchanges
-
-    // Number of pieces source volume could be splitted
-    // (Works like granularity, higly affects gas usage.
-    // Should be called offchain, but could be called onchain
-    // if user swaps not his own funds, but this is still considered
-    // as not safe)
-    // More info: https://github.com/CryptoManiacsZone/1inchProtocol#getexpectedreturn
-    const parts = 2;
-
     const expectedReturn = await this.oneSplitContract.methods
-      .getExpectedReturn(fromToken, toToken, amountWei, parts, flags)
+      .getExpectedReturn(fromToken, toToken, amountWei, this.oneInchParts, this.oneInchFlags)
       .call();
 
     const { returnAmount, distribution } = expectedReturn;
 
     this.logger.debug({
       at: "OneInchExchange",
       message: "GetExpectedReturn",
+      currentTime,
       returnAmount,
       distribution
     });
 
-    // TODO: Remove hardcoded gas
-    const tx = await this.oneSplitContract.methods
-      .swap(fromToken, toToken, amountWei, returnAmount, distribution, flags)
-      .send({ ...options, gasPrice, gas: 8000000 });
+    if (minReturnAmountWei && toBN(returnAmount.toString()).lt(toBN(minReturnAmountWei.toString()))) {
+      this.logger.warn({
+        at: "OneInchExchange",
+        message: "One Inch exchange return amount too low",
+        currentTime,
+        returnAmount,
+        minReturnAmountWei
+      });
+      return;
+    }
+
+    // Swap
+    const swapF = await this.oneSplitContract.methods.swap(
+      fromToken,
+      toToken,
+      amountWei,
+      returnAmount,
+      distribution,
+      this.oneInchFlags
+    );
+    const swapFOptions = { ...options, gasPrice, gas: 8000000, value: fromToken === ETH_ADDRESS ? amountWei : 0 };
+
+    // Gas estimation
+    const GAS_LIMIT_BUFFER = 1.25;
```

Comment:
I would set these as class variables in the constructor

Reply:
Why? I much rather define them as globals that we can import from a constant file. Having everything injected into their respective classes when they're just constant variables seems like unecessary encapsulation to me

Reply:
But you don't import them from a constant file..?

Reply:
I think importing from a constant file is a good idea but you don't do it, you shouldn't make changes to prepare for future refactoring, without leaving detailed comments about your thinking

Reply:
I don't quite get what you're saying...?

Reply:
> I much rather define them as globals that we can import from a constant file

Do you define tem as globals and import them from a constant file?

Reply:
No. What's your point? You wanted a class variable thats going to be referenced once in a single function. Why does this need to be a class variable?

Reply:
It's better practice to keep all constant varrs in one place, in this case in class variables. It will help support further development of additional class functions